### PR TITLE
[WFLY-18644] Removing xerces from distribution

### PR DESCRIPTION
**Pull request by ropalka**
https://issues.redhat.com/browse/WFLY-18644

### CODE REVIEW

**Comment by jbliznak** [comment]
So we have this covered
```
ee-feature-pack/galleon-shared/pom.xml:            <groupId>xerces</groupId>
ee-feature-pack/galleon-shared/pom.xml:            <artifactId>xercesImpl</artifactId>
ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/org/apache/xerces/main/module.xml:<module xmlns="urn:jboss:module:1.9" name="org.apache.xerces">
ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/org/apache/xerces/main/module.xml:        <artifact name="${xerces:xercesImpl}"/>
ee-feature-pack/galleon-shared/src/main/resources/license/licenses.xml:            <groupId>xerces</groupId>
ee-feature-pack/galleon-shared/src/main/resources/license/licenses.xml:            <artifactId>xercesImpl</artifactId>

```

but there are more places using xerces that IMHO should be also removed. If not, why?

```
boms/common-ee/pom.xml:                <groupId>xerces</groupId>
boms/common-ee/pom.xml:                <artifactId>xercesImpl</artifactId>
boms/common-ee/pom.xml:                <version>${version.org.apache.xerces}</version>
pom.xml:        <version.org.apache.xerces>2.12.0.SP05</version.org.apache.xerces>
webservices/tests-integration/pom.xml:            <groupId>xerces</groupId>
webservices/tests-integration/pom.xml:            <artifactId>xercesImpl</artifactId>
testsuite/integration/basic/pom.xml:            <groupId>xerces</groupId>
testsuite/integration/basic/pom.xml:            <artifactId>xercesImpl</artifactId>
```

**Comment by ropalka** [comment]
There are some webservices tests that are using xerces directly @jbliznak . @jimma is there any plan to remove these xerces specific tests from WildFly code base?

**Comment by jbliznak** [comment]
yes they seem to get their own version of xerces

- https://github.com/wildfly/wildfly/blob/main/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/xerces/unit/XercesUsageTestCase.java#L75-L75
- https://github.com/wildfly/wildfly/blob/main/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/xerces/unit/XercesUsageTestCase.java#L105
- https://github.com/wildfly/wildfly/blob/main/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/xerces/ws/unit/XercesUsageInWebServiceTestCase.java#L65-L65

**Comment by jbliznak** [comment]
then this seems just as a wrong comment (the method is actually preparing war and not ear)
- https://github.com/wildfly/wildfly/blob/main/testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/jaxb/unit/JAXBUsageTestCase.java#L49-L49 

and I am not sure if these are relevant anymore but if they are, they seem safe to keep unchanged
- https://github.com/wildfly/wildfly/blob/main/testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/basic/PojoEndpointTestCase.java#L42
- https://github.com/wildfly/wildfly/blob/main/testsuite/integration/ws/src/test/java/org/jboss/as/test/integration/ws/basic/EJBEndpointTestCase.java#L42

**Comment by jimma** [comment]
@ropalka @jbliznak  Except the tests are created to test the xerces jar file packaged in the deployment, the xerces dependency can all be removed.

**Comment by ropalka** [comment]
PR updated as suggested by @jimma @jbliznak

File: testsuite/integration/basic/pom.xml [hunk 1]
```diff
@@ -305,6 +305,7 @@
         <dependency>
             <groupId>xerces</groupId>
             <artifactId>xercesImpl</artifactId>
+            <version>2.12.0.SP05</version>
```
Comment (jbliznak) [inline testsuite/integration/basic/pom.xml:308]: why not just remove this dependency? any other test using this?

Comment (ropalka) [reply testsuite/integration/basic/pom.xml:308]: Yes, there are 2 tests using package import from Xerces in testsuite/integration/basic TS @jbliznak

Comment (ropalka) [reply testsuite/integration/basic/pom.xml:308]: Fixed @jbliznak - now 'provided' scope is used.

**Review by jbliznak — APPROVED** [review]
In terms of desired functionality, this PR is now working as expected.

I will leave for WF guardians if something more is required to tide this up, eg. extract xerces version to property or even manage xerces for tests elsewhere in some test-only BOM (wildfly-standard-test-bom?)

**Review by jimma — APPROVED** [review]

**Pull request merged**